proto/src/dns_handle.rs

```diff
@@ -212,10 +213,11 @@ where
 
     /// creates random query_id, validates against all active queries
     fn next_random_query_id(&self) -> Async<u16> {
+        let between = Range::new(0, u16::max_value());
```

Comment:
This is still off by one, right? The upper bound is exclusive.

Reply:
I read this from their docs: `All the primitive integer types satisfy this property, and the float types normally satisfy it, but rounding may mean high can occur`: https://docs.rs/rand/0.3.18/rand/distributions/range/struct.Range.html

I took that to mean that the top of the Range is the max int, am I reading that incorrectly?

Reply:
The docs confuse me a bit. In one context they discuss primitive integer types satisfying the constraint that it is not inclusive, but then mention `but this may be very difficult`. And then the next sentence discussing potential rounding up to include the max value. If you know better, I'll change it to +1

Reply:
Ok, I reviewed the src for the Rand. It looks like there is no way for max with `u16` to be max. I'll fix this. Thanks for the review.

Reply:
see #292